src/internal/m365/controller.go

```diff
@@ -79,20 +79,24 @@ func NewController(
 		return nil, clues.Wrap(err, "creating api client").WithClues(ctx)
 	}
 
-	rc := resource.UnknownResource
-
-	switch pst {
-	case path.ExchangeService, path.OneDriveService:
-		rc = resource.Users
-	case path.GroupsService:
-		rc = resource.Groups
-	case path.SharePointService:
-		rc = resource.Sites
-	}
+	var rCli *resourceClient
+
+	if pst != path.UnknownService {
```

Comment:
Can't we fail fast for `path.UnknownService`? We are creating a `Controller` for unknown service later in this function. Is this by design?

Reply:
This is by design.  I added a comment for explanation.  Truthfully, it's not my favorite workaround, since it's an ambiguous usage of the UnknownService const.  I originally thought about adding a `NoService` const in the same set, which would allow code to clearly identify the difference between "no service is expected here" and "I should have a service but I don't".  Then I felt uncomfortable about the addition and withdrew it with the intent to bring it up in design discussion.